tests/test_model_class.py

```diff
@@ -11,3 +11,26 @@ class SomeModel(db.Model):
         id = db.Column(db.Integer, primary_key=True)
 
     assert isinstance(SomeModel(), CustomModelClass)
+
+
+def test_repr(db):
+    class User(db.Model):
+        name = db.Column(db.String, primary_key=True)
+
+    class Report(db.Model):
+        id = db.Column(db.Integer, primary_key=True, autoincrement=False)
+        user_name = db.Column(db.ForeignKey(User.name), primary_key=True)
+
+    db.create_all()
+
+    u = User(name='test')
+    db.session.add(u)
```

Comment:
I think create_all, add and flush aren't needed here since you don't need a database roundtrip to access the attributes you just set

Reply:
Unfortunately, `identity` isn't populated until the object is persisted.

Reply:
ah, good point